pkg/ddc/alluxio/ufs_internal.go

```diff
@@ -96,6 +96,156 @@ func (e *AlluxioEngine) shouldMountUFS() (should bool, err error) {
 	return should, err
 }
 
+func (e *AlluxioEngine) getMounts() (resultInCtx []string, resultHaveMounted []string, err error) {
+	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
+	e.Log.V(1).Info("get dataset info", "dataset", dataset)
+	if err != nil {
+		return resultInCtx, resultHaveMounted, err
+	}
+
+	podName, containerName := e.getMasterPodInfo()
+	fileUitls := operations.NewAlluxioFileUtils(podName, containerName, e.namespace, e.Log)
+
+	ready := fileUitls.Ready()
+	if !ready {
+		err = fmt.Errorf("the UFS is not ready")
+		return resultInCtx, resultHaveMounted, err
+	}
+
+	// Check if any of the Mounts has not been mounted in Alluxio
+	for _, mount := range dataset.Spec.Mounts {
+		if common.IsFluidNativeScheme(mount.MountPoint) {
+			// No need for a mount point with Fluid native scheme('local://' and 'pvc://') to be mounted
+			continue
+		}
+		alluxioPathInCtx := utils.UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)
+		resultInCtx = append(resultInCtx, alluxioPathInCtx)
+	}
+
+	// get the mount points have been mountted
+	for _, mount := range dataset.Status.Mounts {
+		if common.IsFluidNativeScheme(mount.MountPoint) {
+			// No need for a mount point with Fluid native scheme('local://' and 'pvc://') to be mounted
+			continue
+		}
+		alluxioPathHaveMountted := utils.UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Status.Mounts)
+		resultHaveMounted = append(resultHaveMounted, alluxioPathHaveMountted)
+	}
+
+	return resultInCtx, resultHaveMounted, err
+
+}
+
+func (e *AlluxioEngine) calculateMountPointsChanges(mountsHaveMountted []string, mountsInContext []string) ([]string, []string) {
```

Comment:
Can we use this? https://github.com/thoas/go-funk/blob/6d0684bc5239f21ff63ae9eb79d21deee99f872d/subtraction.go#L56